compiler/main.bal

```diff
@@ -6,11 +6,12 @@ import wso2/nballerina.comm.diagnostic as d;
 import ballerina/io;
 import ballerina/file;
 
-type CompileError err:Diagnostic|io:Error|file:Error;
+type CompileError err:Diagnostic|io:Error|file:Error|io:Error;
 
 public type Options record {
     boolean testJsonTypes = false;
     boolean showTypes = false;
+    boolean outWat = false;
```

Comment:
shall we change this to `backend` as well.
I am not sure if jBallerina supports unions for CLI options. eg `"llvm"|"wasm" backend = "wasm"` if not, we can do `string backend = "wasm"` and throw an error if it's not matching.